### DISCUSSION TITLE
vscode-json-language-server doesn't support executing code action "Sort JSON"

### DISCUSSION THREAD

**Discussion by mattias-arrelid** [helix-editor/helix#8847] [Troubleshooting]
### Summary

Installing the default language server for JSON as outlined in the official documentation, you get a "Sort JSON" code action in JSON documents. Invoking this results in the following error message:

`Language server does not support executing commands`

I'm not sure why this action would be available by default if it doesn't work?

Output from `hx --health json` below:

```
Configured language servers:
  ✓ vscode-json-language-server: /opt/homebrew/bin/vscode-json-language-server
Configured debug adapter: None
Highlight queries: ✓
Textobject queries: ✘
Indent queries: ✓
```

### Reproduction Steps

 

I tried this:

1. Installed `hx` by invoking `brew install helix`
2. Installed `vscode-langservers-extracted` using `npm`, as indicated is the [ default language server for JSON](https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#json)
3. Opened a JSON document in Helix, `hx test.json`
4. Pressed `Space` followed by `a` (for "Perform code action")
5. Noticed how a popup showed a single item, "Sort JSON"
6. Invoked "Sort JSON"

I expected this to happen:
JSON document would be sorted (by its keys).

Instead, this happened:
You get an error - `Language server does not support executing commands`

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-11-16T21:34:39.019 helix_view::editor [ERROR] editor error: Language server does not support executing commands
2023-11-16T21:37:09.979 helix_view::editor [ERROR] editor error: Language server does not support executing commands
2023-11-16T21:51:18.028 helix_view::editor [ERROR] editor error: Language server does not support executing commands
2023-11-16T21:51:25.082 helix_view::editor [ERROR] editor error: Language server does not support executing commands
2023-11-16T21:52:42.875 helix_view::editor [ERROR] editor error: Language server does not support executing commands
```

</details>

### Platform

macOS

### Terminal Emulator

wezterm 20231107-082518-f0e3eecb

### Installation Method

homebrew

### Helix Version

helix 23.10 (f6021dd0)

**Comment by TornaxO7** [comment]
I can investigate that a bit

**Reply by TornaxO7** [reply]
hm... @arrelid I can't reproduce this. In fact, helix tells me that there are no code actions. May I ask if you have an example file with json?

**Comment by mattias-arrelid** [comment]
Here's a screenshot of a simple JSON document, and me invoking the code actions popup (space + a).

![Screenshot 2023-12-22 at 19 07 26](https://github.com/helix-editor/helix/assets/53868/44dea112-77a7-4d45-bd1b-3bce50f5e914)

Document in question:

```json
{
  "test": {
    "a_key": 1,
    "last_key": [
      "hey",
      "you"
    ],
    "another_key": 2
  }
}
```

If I select everything and pipe to `jq -S`, I get this (which is what I expected the sort json code action would do):

```json
{
  "test": {
    "a_key": 1,
    "another_key": 2,
    "last_key": [
      "hey",
      "you"
    ]
  }
}
```

**Reply by TornaxO7** [reply]
Eh, funny enough I don't get a list of actions:

<details>

![image](https://github.com/helix-editor/helix/assets/50843046/e3545d67-480f-44b6-b8f4-7a5384fa0f56)

</details>

May I ask which `helix` and `vscode-json-language-server` version you're using?

**Reply by mattias-arrelid** [reply]
```
$ hx --version
helix 23.10 (f6021dd0)

$ brew info vscode-langservers-extracted
==> vscode-langservers-extracted: stable 4.8.0 (bottled)
Language servers for HTML, CSS, JavaScript, and JSON extracted from vscode
https://github.com/hrsh7th/vscode-langservers-extracted
/opt/homebrew/Cellar/vscode-langservers-extracted/4.8.0 (4,557 files, 74.6MB) *
  Poured from bottle using the formulae.brew.sh API on 2023-11-28 at 21:04:46
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/v/vscode-langservers-extracted.rb
License: MIT
==> Dependencies
Required: node ✔
==> Analytics
...
```

**Reply by TornaxO7** [reply]
may I ask for `hx --health json`?

**Reply by mattias-arrelid** [reply]
```
$ hx --health json
Configured language servers:
  ✓ vscode-json-language-server: /opt/homebrew/bin/vscode-json-language-server
Configured debug adapter: None
Highlight queries: ✓
Textobject queries: ✘
Indent queries: ✓
```

**Reply by TornaxO7** [reply]
ok, I can't see the version of your LSP json-lsp. May I ask if you can try finding this out?
Just verify that its version is [1.3.4](https://www.npmjs.com/package/vscode-json-languageserver).

**Reply by mattias-arrelid** [reply]
Just so that I don't forget to mention it: I installed [vscode-langservers-extracted](https://www.npmjs.com/package/vscode-langservers-extracted) instead of [vscode-json-languageserver](https://www.npmjs.com/package/vscode-json-languageserver) to get the latest version of the JSON LSP (as you can see in the npm package you link to, that standalone version is ~3 years old).

Looking at [package.json](https://github.com/hrsh7th/vscode-langservers-extracted/blob/master/package.json) for `vscode-json-languageservice`, it's got its version set to `^5.3.7`.

**Reply by mattias-arrelid** [reply]
...which seems to correspond to the (tagged) version that MS published:
https://github.com/microsoft/vscode-json-languageservice/tags

**Reply by TornaxO7** [reply]
Ah, so it looks like that I'm using the wrong one.

**Reply by mattias-arrelid** [reply]
Not necessarily wrong - just a quite old one :)

**Comment by mattias-arrelid** [comment]
@TornaxO7 Please see the attached example
